g11n-ws/modules/md-data-api-s3impl/src/main/java/com/vmware/vip/messages/data/dao/impl/S3ProductDaoImpl.java

```diff
@@ -121,7 +124,45 @@ public String getVersionInfo(String productName, String version) throws DataExce
       return result;
    }
 
-   public Map<String, String[]> getProductsAndVersions() {
-      return null;
-   }
+/* (non-Javadoc) * @see com.vmware.vip.messages.data.dao.api.IProductDao#getVersionList(java.lang.String) */
+@Override
+public List<String> getVersionList(String productName) throws DataException {
+    // TODO Auto-generated method stub
+    String basePath = S3Utils.S3_L10N_BUNDLES_PATH+productName +  ConstantsChar.BACKSLASH;
+    ListObjectsV2Result versionListResult = s3Client.getS3Client().listObjectsV2(config.getBucketName(),basePath);
+    
+    if(versionListResult != null) {
+        List<S3ObjectSummary> versionListSummary = versionListResult.getObjectSummaries();
+        Set<String> versionset = new HashSet<>();
+        for (S3ObjectSummary s3productName : versionListSummary) {
+            versionset.add(s3productName.getKey().replace(basePath, "").split(ConstantsChar.BACKSLASH)[0]);
+          }
+        List<String> result = new ArrayList<>();
+        for(String version: versionset) {
+            result.add(version);
+        }
+      return result;
+      
+    }else {
+     throw new DataException(productName + " no available version in s3");
+    }
+}
+
+  /* public Map<String, String[]> getProductsAndVersions() {
```

Comment:
If this code is no longer used, it's better to delete it.

Reply:
done